app/javascript/app/service-provider-form.js

```diff
@@ -59,35 +75,45 @@ $(function () {
         returnToSpUrl.setAttribute('required', 'required');
         break;
       default:
-        samlFields.show();
-        oidcFields.show();
-    }
-  };
-
-  const toggleIALOptions = (ial) => {
-    switch (ial) {
-      case '1':
-        failureToProofURL.hide();
-        toggleIAL1Options();
-        break;
-      case '2':
-        failureToProofURL.show();
-        toggleIAL2Options();
-        break;
-      default:
-        failureToProofURL.show();
-        toggleIAL2Options();
+        samlFields.forEach(field => field.classList.remove('display-none'));
+        oidcFields.forEach(field => field.classList.remove('display-none'));
     }
   };
 
   // Page initialization
-  toggleFormFields(idProtocol.filter(':checked').val());
-  toggleIALOptions(ialLevel.val());
+  toggleFormFields(activeIdProtocol.value);
 
   // Event triggers
-  idProtocol.change((evt) => toggleFormFields(evt.target.value));
+  idProtocols.forEach(idProtocol => {
+    idProtocol.addEventListener("change", (event) => toggleFormFields(event.target.value));
+  });
+};
+
+function serviceProviderForm() {
+  ialOptionSetup();
+  protocolOptionSetup();
+  certificateUploadSetup();
+}
+
+window.addEventListener('DOMContentLoaded', serviceProviderForm);
 
-  ialLevel.change((evt) => toggleIALOptions(evt.target.value));
+$(function () {
+  if (!$('.service-provider-form').length) {
+    return;
+  }
```

Comment:
the PR title is "start converting" so this is sticking around for future work?

Reply:
Yeah, I'm gonna be out next week so I wanted to get the progress I've made up